### PR TITLE
DEV: Make prompts available on `CurrentUserSerializer`

### PR DESCRIPTION
**Pull request by keegangeorge**
Instead of making a request to `/discourse-ai/prompts.json` to fetch the list of available prompts for the AI Helper, we ship the prompts automatically with the `CurrentUserSerializer` so its available without any delay. Prompts are also cached.

### CODE REVIEW

File: lib/ai_helper/entry_point.rb [hunk 1]
```diff
@@ -19,6 +19,20 @@ def inject_into(plugin)
...
+        plugin.add_to_serializer(
+          :current_user,
+          :ai_helper_prompts,
+          include_condition: -> do
+            SiteSetting.composer_ai_helper_enabled && scope.authenticated? &&
+              scope.user.in_any_groups?(SiteSetting.ai_helper_allowed_groups_map)
+          end,
+        ) do
+          ActiveModel::ArraySerializer.new(
+            DiscourseAi::AiHelper::Assistant.new.available_prompts,
```
Comment (SamSaffron) [inline lib/ai_helper/entry_point.rb:32]: this is great for starters, but I think we can eliminate this query to the database with a cache, feels kind of important cause there is no need to run through all of this work on every page load for users with access.

**Review by SamSaffron — APPROVED** [review]

File: lib/ai_helper/entry_point.rb (hunk 1 above)
Comment (SamSaffron) [reply lib/ai_helper/entry_point.rb:32]: not blocking can happen in a followup pr.

**Comment by SamSaffron** [comment]
we are fiddling with the current user serializer, so we should at least ensure the code is covered and that it is rendered in a spec.  I think simplest is to make a get request to current user and then you have it all in JSON format

File: lib/ai_helper/assistant.rb [hunk 2]
```diff
@@ -3,35 +3,40 @@
 module DiscourseAi
   module AiHelper
     class Assistant
-      def available_prompts(name_filter: nil)
-        cp = CompletionPrompt
-        prompts = []
+      AI_HELPER_PROMPTS_CACHE_KEY = "ai_helper_prompts"
+
+      def available_prompts
+        prompts = Discourse.cache.read(AI_HELPER_PROMPTS_CACHE_KEY)
```
Comment (xfalcox) [inline lib/ai_helper/assistant.rb:9]: This pattern of `return if it's on cache, otherwise fetch it, is already implemented by Discourse.cache.fetch

**Review by romanrizzi — APPROVED** [review]

**Pull request merged**